docs/source/common/optimizers.rst

```diff
@@ -67,27 +67,91 @@ Here is the same example as above using a ``closure``.
             opt.zero_grad()
 
 
-.. code-block:: python
-
-    # Scenario for a GAN.
-    def training_step(...):
-        opt_gen, opt_dis = self.optimizers()
+.. tip:: Be careful where you call ``zero_grad`` or your model won't converge. It is good pratice to call ``zero_grad`` before ``manual_backward``.
 
-        # compute generator loss
-        loss_gen = self.compute_generator_loss(...)
-
-        # zero_grad needs to be called before backward
-        opt_gen.zero_grad()
-        self.manual_backward(loss_gen)
-        opt_gen.step()
 
-        # compute discriminator loss
-        loss_dis = self.compute_discriminator_loss(...)
+.. code-block:: python
```

Comment:
```suggestion
.. testcode::
```

Reply:
Not sure it makes sense to have test code if the code contains ... for some part right ?

Reply:
contains what? a not long time ago we had PRs with fixing python syntax in docs and these tests do not cost anything and catches these kinds of issues